week_1/String/find_all_anagrams/find_all_anagrams.js

```diff
@@ -0,0 +1,67 @@
+// Given two strings long and short, return an array of all the start indices of short's anagrams in long. You may return the answer in any order.
+/**
+ * @param {string} long A longer string
+ * @param {string} short A shorter string
+ * @return {number[]}
+ */
+
+const findAllAnagrams = (long, short) => {
+	if (long.length < short.length) return [];
+
+	const indices = [];
+	// Create a Map that will contain all the values of the shorter string
+	const shortMap = new Map();
+	for (char of short) {
+		shortMap.set(char, shortMap.get(char) + 1 || 1);
+	}
+	console.log(shortMap);
+
+	// Build up a map of equal length from the beginning of the longer string
+	const longMap = new Map();
+	for (let i = 0; i < short.length; i++) {
+		let char = long[i];
+		longMap.set(char, longMap.get(char) + 1 || 1);
+	}
+
+	// Then, if they're the same, add 0 to the array of indices.
+	// To me, it's more intuitive to do this in reverse, because if one value is unequal, it doesn't work,
```

Comment:
I'll often leave these types of personal explanations to the PR itself. After opening a PR, I'll come in a leave comments throughout so the reviewer can understand my thought process. This type of comment doesn't belong in the codebase, though